controllers/machineset_controller.go

```diff
@@ -192,6 +192,14 @@ func (r *MachineSetReconciler) reconcile(ctx context.Context, cluster *clusterv1
 	}
 
 	// Make sure selector and template to be in the same cluster.
+	if machineSet.Spec.Selector.MatchLabels == nil {
+		machineSet.Spec.Selector.MatchLabels = make(map[string]string)
+	}
+
```

Comment:
why is this change in this PR?

Reply:
+1

Reply:
seems like they are two different commits with two separate descriptions in PR description

Reply:
I'll open a separate PR for the second commit so we can keep it cleaner.